NLSY/nlsy_imputation.qmd

```diff
@@ -0,0 +1,227 @@
+        race_eth = ifelse(hisp == "Yes" & race == "American Indian, Eskimo, or Aleut", "NonBW Hispanic", race_eth),
+        race_eth = ifelse(hisp == "Yes" & race == "Asian or Pacific Islander", "NonBW Hispanic", race_eth),
+        race_eth = ifelse(hisp == "Yes" & race == "Something else? (SPECIFY)", "NonBW Hispanic", race_eth),
+        race_eth = ifelse(hisp == "No" & race == "American Indian, Eskimo, or Aleut", "NonBW NonHispanic", race_eth),
+        race_eth = ifelse(hisp == "No" & race == "Asian or Pacific Islander", "NonBW NonHispanic", race_eth),
+        race_eth = ifelse(hisp == "No" & race == "Something else? (SPECIFY)", "NonBW NonHispanic", race_eth),
+        race_eth = factor(race_eth))
+
+test$ret_sav <- ifelse(test$savings == "YES", 1, 0)
+test$hown <- ifelse(test$home == "OWNS OR IS BUYING; LAND CONTRACT", 1, 0)
+test$both_pars <- ifelse(test$both_parents == "Yes", 1, 0)
+test$par_dec <- ifelse(test$par1_deceased == "Yes" | test$par2_deceased == "Yes", 1, 0) # Either parent deceased
+
+```
+
+```{r}
+# Looking for significance with non-missing income and wealth
+non_missing <- test %>%
+    filter(!is.na(pincome),
+           !is.na(pnetworth))
+
+summary(model <- lm(pincome ~ mom_age + hisp + race + mom_educ_hs + dad_educ_hs + ret_sav + hown + both_pars + par_dec, data = non_missing))
+#Non-significant: par_dec - dropping for now
+```
+
+### Checking for multi-collinearity - correlation plot for numeric variables
+
+```{r}
+
+testing_cor <- test %>%
+    select(-id, -hispanic, -race, -mom_education, -dad_education, -savings, -home, -both_parents, -par1_deceased, -par2_deceased, - mom_age_birth, -pincome, -pnetworth, -par_dec, -mom_educ_hs,-dad_educ_hs, -hisp,-race_eth) %>%
+    drop_na()
+
+M <- cor(testing_cor)
+corrplot(M, type = "upper")
+```
+
+### Checking for multi-collinearity - VIF and Successive addition of regressors
+
+```{r}
+
+summary(model <- lm(pincome ~ mom_age, data = test)) #Adj R2 = 0.057
+summary(model <- lm(pincome ~ mom_age + hisp, data = test)) #Adj R2 = 0.09
+car::vif(model)
+summary(model <- lm(pincome ~ mom_age + hisp + race, data = test)) # #Adj R2 = 0.157
+car::vif(model)
+
+summary(model <- lm(pincome ~ mom_age + hisp + race + mom_educ_hs, data = test)) #Adj R2 = 0.1899
+car::vif(model)
+
+summary(model <- lm(pincome ~ mom_age + hisp + race + mom_educ_hs + dad_educ_hs, data = test)) #Adj R2 = 0.1616
+car::vif(model)
+
+summary(model <- lm(pincome ~ mom_age + hisp + race + mom_educ_hs + dad_educ_hs + ret_sav, data = test)) #Adj R2 = 0.199
+car::vif(model)
+
+summary(model <- lm(pincome ~ mom_age + hisp + race + mom_educ_hs + dad_educ_hs + ret_sav + hown, data = test)) #Adj R2 = 0.2107
+car::vif(model)
+
+summary(model <- lm(pincome ~ mom_age + hisp + race + mom_educ_hs + dad_educ_hs + ret_sav + hown + both_pars, data = test)) #Adj R2 = 0.2108
+car::vif(model)
+```
+
+## Checking for missings in the predictor variables
+
```

Comment:
Same here, some explanation would be helpful. What does this ration mr/(mr+mm) tell us and how do we use it to make a decision about the imputation process? For example, (race,  pincome) is 0.6, and (pincome, race) is 0.987. Does this mean that when pincome is missing, race is also missing in 60% of cases, and when race is missing, pincome is also missing in 98.7% of cases?

Reply:
The interpretation of mr/(mr+mm) is "percentage of usable cases to impute row variable from column variable." So in the example above, a ratio of 0.6 means that for 60% of observations with non-missing race, pincome is also non-missing.

Documentation here for reference: https://www.rdocumentation.org/packages/mice/versions/3.16.0/topics/md.pairs

Reply:
The interpretation of mr/(mr+mm) is "percentage of usable cases to impute row variable from column variable." So in the example above, a ratio of 0.6 means that for 60% of observations with missing race, pincome is non-missing and can thus be used in imputation

Documentation here for reference: https://www.rdocumentation.org/packages/mice/versions/3.16.0/topics/md.pairs

Reply:
It's a non-intuitive measure, but it's the one used in the documentation. As Judah shared, a lower value indicates higher missingness in the predictor for the outcome of interest.